kernel/thread.c

```diff
@@ -411,6 +412,9 @@ static tcb_t *thread_select(tcb_t *parent)
 	if (thr == NULL)
 		return NULL;
 
+	if ((thr = rr_select()) != NULL)
```

Comment:
If `rr_select` returns NULL, `thr`will be modified as NULL and unexpected result
will happen in the following code.